### PR TITLE
FOGL 5891

### PR DESCRIPTION
**Pull request by deepanshu-yadav**
TODO

1. Change documentation for Fledge.
2. Change documentation for flir and FogLAMP.
3. Make changes in system tests of CSV writer filter.
4. Make changes in system tests of this plugin.

### CODE REVIEW

File: python/fledge/plugins/south/csvplayback/csvplayback.py [hunk 1]
```diff
@@ -362,15 +466,62 @@ def __init__(self, handle):
...
+        self.meta_data = {}
+        self.current_csv_file = None
+        self.shutdown_plugin = False
         self.read_csv_file()
 
+    def get_csv_file_name(self):
+        if self.handle['csvDirName']['value'] == "FLEDGE_DATA":
+            csv_dir = _FLEDGE_DATA
+        else:
+            csv_dir = _FLEDGE_DATA
```
Comment (praveen-garg) [inline python/fledge/plugins/south/csvplayback/csvplayback.py:481]: How it is different than if?

File: python/fledge/plugins/south/csvplayback/csvplayback.py [hunk 2]
```diff
@@ -101,15 +173,16 @@
...
         'minimum': '1',
-        'order': '6'
+        'order': '14'
     },
     'timestampStyle': {
         'description': 'Select "continuous" mode asset timestamp processing style.',
         'type': 'enumeration',
         'default': 'current time',
+        'validity': "variableCols == \"false\"",
         'options': ['current time', 'copy csv value', 'move csv value', 'use csv sample delta'],
```
Comment (praveen-garg) [inline python/fledge/plugins/south/csvplayback/csvplayback.py:110]: Keep options value the same way for others i.e no snake_case or camelCase. Here make it Current time, Copy..., Move..., Use...

File: python/fledge/plugins/south/csvplayback/csvplayback.py [hunk 3]
```diff
@@ -62,28 +63,98 @@
...
-    'csvFilename': {
-        'description': 'CSV file name with extension found in FLEDGE_ROOT/data',
+    'csvDirName': {
+        'description': 'The directory where CSV file exists. Default is '
+                       'FLEDGE_DATA or FLEDGE_ROOT/data',
         'type': 'string',
-        'default': '',
-        'displayName': 'CSV file name',
+        'default': 'FLEDGE_DATA',
+        'displayName': 'CSV Directory Name',
```
Comment (praveen-garg) [inline python/fledge/plugins/south/csvplayback/csvplayback.py:71]: CSV directory name (consistent case for all display name)

File: python/fledge/plugins/south/csvplayback/csvplayback.py [hunk 4]
```diff
@@ -505,11 +753,20 @@ def chunk_to_readings(self, chunk):
                 modified_timestamp = str(now_timestamp.replace(microsecond=useconds))
                 useconds += uniform_interval
 
-            reading = {
-                'asset': self.asset_name,
-                'timestamp': modified_timestamp,
-                'readings': row_values
-            }
+            if not self.meta_data_ingested and self.process_metadata:
```
[the remark this answers is not in the document]
Comment (deepanshu-yadav) [reply python/fledge/plugins/south/csvplayback/csvplayback.py:756]: Ok doing so.

**Comment by deepanshu-yadav** [comment]
Tests are passing. Merging this branch. Documentation will be added in other branch named fogl-5891.docs.

**Pull request merged**